app/src/main/java/com/epam/caloriecalc/ui/addnew/AddItemViewModel.kt

```diff
@@ -0,0 +1,62 @@
+package com.epam.caloriecalc.ui.addnew
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.epam.caloriecalc.data.local.entities.IntakeRecord
+import com.epam.caloriecalc.data.local.repository.CalorieRepository
+import com.epam.caloriecalc.util.AddItemEvent
+import com.epam.caloriecalc.util.SnackbarActionType
+import com.epam.caloriecalc.util.UiEvent
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class AddItemViewModel @Inject constructor(
+    internal val repository: CalorieRepository
+) : ViewModel() {
+
+    init {
+        viewModelScope.launch {
+            for (item in testitems)
+                repository.insertProduct(item)
+        }
+    }
+
+    private lateinit var pendingItem: IntakeRecord
```

Comment:
Can this be converted into a local variable?

Reply:
sure